src/static/js/libs/gamepad.js

```diff
@@ -475,18 +484,53 @@
       return Gamepad.Type.UNSUPPORTED;
     }
   };
+
+  /* Periodically scan for gamepads for Chrome support */
+  Gamepad.prototype._scanGamepads = function() {
+    var gamepads = navigator.getGamepads ? navigator.getGamepads() : (navigator.webkitGetGamepads ? navigator.webkitGetGamepads() : []);
+
+    for(let i = 0; i < gamepads.length; ++i)
+    {
+      let gamepad = gamepads[i];
+
+      //If we get a valid gamepad
+      if(gamepad != null)
+      {
+        //If this gamepad isn't already registered
+        if(this.gamepads[i] == null)
+        {
+          //Register it
+          this._connect(gamepad);
+        }
+      }
+      else
+      {
+        //If there is a gamepad registered in this spot, we need to deregister (it has been removed)
+        if(this.gamepads[i] != null)
+        {
+          gamepad = this.gamepads[i];
+          this._disconnect(gamepad);
+        }
+      }
+    }
+  }
+
+
   /**
  * Updates the controllers, triggering TICK events.
  */
   Gamepad.prototype._update = function () {
     var self = this, controlName, isDown, lastDown, downBtnIndex, mapping, value, i, j;
+
+    //Support chrome, scan for gamepads
+    this._scanGamepads();
     switch (this.platform) {
```

Comment:
I believe the entire switch statement can be removed as the _scanGamepads() that you added duplicates the same feature of polling the gamepad API for available gamepads.